src/System.Runtime.Extensions/src/System/Environment.Win32.Unix.cs

```diff
@@ -21,12 +22,23 @@ public static string GetEnvironmentVariable(string variable, EnvironmentVariable
 
         public static IDictionary GetEnvironmentVariables()
         {
-            return EnvironmentAugments.GetEnvironmentVariables();
+            // To maintain complete compatibility with prior versions we need to return a Hashtable.
+            // We did ship a prior version of Core with LowLevelDictionary, which does iterate the
+            // same (e.g. yields DictionaryEntry), but it is not a public type.
+            //
+            // While we could pass Hashtable back from CoreCLR the type is also defined here. We only
+            // want to surface the local Hashtable.
+            return new Hashtable(EnvironmentAugments.GetEnvironmentVariables());
         }
 
         public static IDictionary GetEnvironmentVariables(EnvironmentVariableTarget target)
         {
-            return EnvironmentAugments.GetEnvironmentVariables(target);
+            // See comments in GetEnvironmentVariables()
+            IDictionary values = EnvironmentAugments.GetEnvironmentVariables(target);
+            if (values == null)
```

Comment:
It returns empty dictionary: https://github.com/dotnet/coreclr/blob/e06afa73cc284c6659a03eb475eb426d08938e77/src/mscorlib/src/System/Environment.Unix.cs#L54 no?